lua/astrocommunity/pack/typescript/init.lua

```diff
@@ -49,7 +49,19 @@ return {
   },
   {
     "williamboman/mason-lspconfig.nvim",
-    opts = function(_, opts) opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "tsserver") end,
+    opts = function(_, opts)
+      opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "tsserver", "eslint-lsp")
+    end,
+    init = function()
+      vim.api.nvim_create_autocmd("BufWritePost", {
+        desc = "Fix all eslint errors",
+        pattern = { "*.tsx", "*.ts", "*.jsx", "*.js" },
+        group = "...",
+        callback = function()
+          if vim.fn.exists ":EslintFixAll" > 0 then vim.cmd "EslintFixAll" end
+        end,
+      })
+    end,
```

Comment:
```suggestion
    opts = function(_, opts)
      opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "tsserver", "eslint")

      vim.api.nvim_create_autocmd("LspAttach", {
        group = vim.api.nvim_create_augroup("eslint_fix_creator", { clear = true }),
        desc = "Create autocommand in buffers where eslint attaches",
        callback = function(args)
          if assert(vim.lsp.get_client_by_id(args.data.client_id)).name == "eslint" then
            vim.api.nvim_create_autocmd("BufWritePost", {
              desc = "Fix all eslint errors",
              buffer = args.buf,
              group = vim.api.nvim_create_augroup(("eslint_fix_%d"):format(args.buf), { clear = true }),
              callback = function() pcall(vim.cmd.EslintFixAll) end,
            })
          end
        end,
      })
    end,
```

Suggestion does a few things:

- We only need to create auto commands for `:EslintFixAll` for buffers that have `eslint` attached.
- Makes it easier to disable and also `init` functions do not stack so that's a bad insertion point
- fixes bug `eslint-lsp` should be `eslint` since that's the name of the language server

Reply:
This also decreases the maintenance burden of maintaining a separate list of filetypes that eslint may or may not use. and rather relies on the filetype detection of the language server attaching.

Reply:
actually this is wrong, let me revise the suggestion. Apparently the `pcall` here is not ideal